src/contracts/pods/EigenPod.sol

```diff
@@ -443,30 +542,75 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
         external
         onlyEigenPodManager
     {
-        // reduce the restakedExecutionLayerGwei
-        restakedExecutionLayerGwei -= uint64(amountWei / GWEI_TO_WEI);
-
         emit RestakedBeaconChainETHWithdrawn(recipient, amountWei);
-
         // transfer ETH from pod to `recipient`
         _sendETH(recipient, amountWei);
     }
 
+
+    /**
+     * @notice Called by the pod owner to activate restaking by withdrawing 
+     * all existing ETH from the pod and preventing further withdrawals via 
+     * "withdrawBeforeRestaking()"
+    */ 
+    function activateRestaking() external onlyWhenNotPaused(PAUSED_EIGENPODS_VERIFY_CREDENTIALS) onlyEigenPodOwner hasNeverRestaked {
+        hasRestaked = true;
+        _processWithdrawalBeforeRestaking(podOwner);
+    }
+
     /// @notice Called by the pod owner to withdraw the balance of the pod when `hasRestaked` is set to false
     function withdrawBeforeRestaking() external onlyEigenPodOwner hasNeverRestaked {
-        mostRecentWithdrawalBlockNumber = uint32(block.number);
-        _sendETH(podOwner, address(this).balance);
+        _processWithdrawalBeforeRestaking(podOwner);
+    }
+
+    function validatorPubkeyHashToInfo(bytes32 validatorPubkeyHash) external view returns(ValidatorInfo memory) {
+        return _validatorPubkeyHashToInfo[validatorPubkeyHash];
+    }
+
+    function validatorStatus(bytes32 pubkeyHash) external view returns (VALIDATOR_STATUS) {
+        return _validatorPubkeyHashToInfo[pubkeyHash].status;
     }
 
     // INTERNAL FUNCTIONS
     function _podWithdrawalCredentials() internal view returns(bytes memory) {
         return abi.encodePacked(bytes1(uint8(1)), bytes11(0), address(this));
     }
 
+    function _processWithdrawalBeforeRestaking(address podOwner) internal {
+        mostRecentWithdrawalTimestamp = uint32(block.timestamp);
+        _sendETH(podOwner, address(this).balance);
+    }
+
     function _sendETH(address recipient, uint256 amountWei) internal {
         delayedWithdrawalRouter.createDelayedWithdrawal{value: amountWei}(podOwner, recipient);
     }
 
+    function _calculateRestakedBalanceGwei(uint64 amountGwei) internal view returns (uint64){
+        if (amountGwei <= RESTAKED_BALANCE_OFFSET_GWEI) {
+            return 0;
+        }
+        /**
+        * calculates the "floor" of amountGwei - RESTAKED_BALANCE_OFFSET_GWEI.  By using integer division 
+        * (dividing by GWEI_TO_WEI = 1e9 and then multiplying by 1e9, we effectively "round down" amountGwei to 
+        * the nearest ETH, effectively calculating the floor of amountGwei.
```

Comment:
specifically, this is the 'floor' because the input is *expected* to be an effective balance, which should only be a whole number of ETH

Reply:
also going to add that the usage of "nearest ETH" and "floor" here implicitly presume the value of `RESTAKED_BALANCE_OFFSET_GWEI`. If we are going that far, we probably just want to hard-code the value rather than having it be an immutable variable?

Reply:
yeah good point, I think its better to keep it immutable but may want to consider what implications that may have on this function if things change in the future.